### ISSUE TITLE
Prepare the recipe "sharing-rules" part of the collection "global-view" error message - Oxygen version

### ISSUE THREAD

**Issue by EricBURTE** [SalesforceLabs/OrgCheck#724]
**Describe the bug**
When launching a global view, there was an error popping ùmentioning "Oops we had an issue... (Prepare the recipe "sharing-rules" part of the collection "global-view")"

Error #0: UNKNOWN_EXCEPTION: An unexpected error occurred. Please include this ErrorId if you contact support: 1764139461-519855 (-1127051950)
{
  "name": "sf:UNKNOWN_EXCEPTION",
  "errorCode": "sf:UNKNOWN_EXCEPTION",
  "contextInformation": {
    "Alias": "sharing-rules",
    "Cause": "UNKNOWN_EXCEPTION: An unexpected error occurred. Please include this ErrorId if you contact support: 1764139461-519855 (-1127051950)",
    "Where": "RecipeManagerImpl._prepareRecipe"
  }
}

**To Reproduce**
Click on sub menu "Salesforce Organization" > " Global View"

<img width="1321" height="388" alt="Image" src="https://github.com/user-attachments/assets/8be98534-9340-4576-aa70-19ed6b765aca" />

**Comment by VinceFINET** [comment]
Its an issue that was identified by the salesforce internal team.
it happens when you try to get information of a sharing rule from the metadata api for an unknown sobject.

**Comment by VinceFINET** [comment]
it would be interesting to know which object causes this.
i suspect that i am looking for a sharing rule for an object without it's package name or similar...

**Comment by VinceFINET** [comment]
@EricBURTE  Is the issue still there?

**Comment by Sanket-git-14** [comment]
Even I got the same exception in **Salesforce Organization > Global view** and **Data model > Sharing Rules** (attached screenshot)

The following NPSP objects causes the exception: 
1. npsp__RecurringDonationSchedule__c
2. npsp__Payment_Services_Configuration__c
3. npsp__AutoNumber__c

<img width="1327" height="442" alt="Image" src="https://github.com/user-attachments/assets/be5189ae-b583-460f-a18c-e55957150a42" />

**Comment by VinceFINET** [comment]
so need to fix this and check if it's specific to sharing rules from custom object coming from a package namespace.

**Comment by VinceFINET** [comment]
Hello i finally have some time to work on this one.
so i created a dev org with npsp and had that exact error (same GACK error in parenthesis I mean):

<img width="1573" height="669" alt="Image" src="https://github.com/user-attachments/assets/b66babf7-fab9-4eba-aa3e-acfc742f2d09" />

**Comment by VinceFINET** [comment]
In this example, org check is retrieving sharing rules information via the metadata api. Some of the http queries were failing with that error. The body of these ones were:
```
<readMetadata>
  <type>SharingRules</type>
  <fullNames>npsp__Address_Verification_Settings__c</fullNames>
  <fullNames>npsp__Allocations_Settings__c</fullNames>
  <fullNames>npsp__AutoNumber__c</fullNames>
  <fullNames>npsp__Batch_Data_Entry_Settings__c</fullNames>
  <fullNames>npsp__Custom_Column_Header__c</fullNames>
  <fullNames>npsp__Custom_Notification__c</fullNames>
  <fullNames>npsp__Customizable_Rollup_Settings__c</fullNames>
  <fullNames>npsp__Data_Import_Field_Mapping__c</fullNames>
  <fullNames>npsp__Data_Import_Settings__c</fullNames>
  <fullNames>npsp__DeploymentEvent__c</fullNames>
</readMetadata>
<readMetadata>
  <type>SharingRules</type>
  <fullNames>npsp__Package_Settings__c</fullNames>
  <fullNames>npsp__Payment_Services_Configuration__c</fullNames>
  <fullNames>npsp__RecurringDonationStatusMapping__c</fullNames>
  <fullNames>npsp__Rollup__c</fullNames>
  <fullNames>npsp__Schedulable__c</fullNames>
  <fullNames>npsp__Trigger_Handler__c</fullNames>
  <fullNames>npsp__fflib_AppBinding__c</fullNames>
  <fullNames>npsp__Fund__c</fullNames>
  <fullNames>npsp__Relationship_Sync_Excluded_Fields__c</fullNames>
  <fullNames>WorkPlan</fullNames>
</readMetadata>
<readMetadata>
  <type>SharingRules</type>
  <fullNames>npe03__Recurring_Donations_Settings__c</fullNames>
  <fullNames>npe4__Relationship_Error__c</fullNames>
  <fullNames>npe4__Relationship_Lookup__c</fullNames>
  <fullNames>npe4__Relationship_Settings__c</fullNames>
  <fullNames>npe5__Affiliations_Settings__c</fullNames>
  <fullNames>npsp__Account_Soft_Credit__c</fullNames>
  <fullNames>npsp__Partial_Soft_Credit__c</fullNames>
  <fullNames>npsp__RecurringDonationChangeLog__c</fullNames>
  <fullNames>npsp__RecurringDonationSchedule__c</fullNames>
  <fullNames>npsp__Addr_Verification_Settings__c</fullNames>
</readMetadata>
```

I tried these batchs separately and the object that have issues are:
- npsp__RecurringDonationSchedule__c 
- npsp__AutoNumber__c 

Indeed these are not returned by EntityDefinition using tooling api
```
SELECT Id, NamespacePrefix, DeveloperName 
FROM EntityDefinition 
WHERE DurableId   IN ('npe4__Relationship_Settings__c', 'npsp__RecurringDonationSchedule__c', 'npsp__AutoNumber__c')
```

But they are returned by CustomObject 
```
SELECT Id, NamespacePrefix, DeveloperName 
FROM CustomObject 
WHERE DeveloperName IN ('Relationship_Settings', 'RecurringDonationSchedule', 'AutoNumber')
```

**Comment by VinceFINET** [comment]
The code initially listed the custom objects that have a specified sharing model using this query:
```
SELECT NamespacePrefix, DeveloperName 
FROM CustomObject 
WHERE SharingModel != ''
```

We need to switch to EntityDefinition because the result of the previous query could not be supported by the metadata api, so the new query should look like:
```
SELECT DurableId
FROM EntityDefinition 
WHERE InternalSharingModel != ''
```

BUT wait a minute, what if i don't have to list the custom object out of metadata api? Normally when setting "*" you let the api search all the entity instead??

Removing the query and just call the metadata api with "*".
